presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/query/PinotFilterExpressionConverter.java

```diff
@@ -171,9 +297,28 @@ public PinotExpression visitCall(CallExpression call, Function<VariableReference
                 return handleLogicalBinary(operatorType.getOperator(), call, context);
             }
         }
+        // This is the handling to allow queries like `millisSinceEpoch < 1391126400000` been handled correctly.
+        // Otherwise TypeManager.canCoerce(...) will return false and directly fail this query.
+        if (functionMetadata.getName().getFunctionName().equalsIgnoreCase("$literal$timestamp") ||
+                    functionMetadata.getName().getFunctionName().equalsIgnoreCase("$literal$date")) {
+            return handleDateAndTimestamp(call, context);
```

Comment:
I am not sure where my previous comment went, but should we rename this method to handleDateAndTimestampMagicLiteralFunction ? (Magic literal function is how presto refers to them ... see `LiteralEncoder`)